bug: Astro build fails with Tailwind CSS 4.2.2 ESM chunk resolution error

**Found during:** Build testing (2026-04-01)
**Severity:** high (blocks deployment)

## Error
```
Cannot find module '.../dist/.prerender/chunks/_.._CFlRqXUi.mjs'
imported from '.../dist/.prerender/chunks/prerender_DFul5d5u.mjs'
```

Originates from `@tailwindcss/node@4.2.2` ESM cache loader (`esm-cache.loader.mjs`).

## Environment
- Astro 6.1.x, Tailwind CSS 4.2.2, `@tailwindcss/vite`
- Node.js 22.x, pnpm 9.x
- Occurs even with `NODE_OPTIONS='--max-old-space-size=8192'`
- `rm -rf dist/` fails (directory not empty — file locks during build)

## Likely Cause
The Tailwind CSS 4 Vite plugin generates prerender chunks with content-hashed filenames. During large static site builds (53 titles, 1000+ pages), the ESM module cache gets stale or the chunk is GC'd before all prerender workers reference it.

## Potential Fixes
1. Pin `@tailwindcss/node` to 4.1.x (before ESM cache loader change)
2. Increase Node.js `--max-old-space-size` further (16GB+)
3. Split build into smaller batches by title
4. Report upstream to tailwindlabs/tailwindcss
5. Try `@tailwindcss/postcss` instead of `@tailwindcss/vite` for build stability

## Workaround
Package tests (248 tests, 0 failures) all pass. Only the static site generation is affected.

--- a/apps/web/package.json
+++ b/apps/web/package.json
@@ -18,11 +18,11 @@
     "@fontsource/public-sans": "^5.2.7",
     "@octokit/rest": "^22.0.1",
     "@tailwindcss/typography": "^0.5.19",
-    "@tailwindcss/vite": "^4.2.2",
+    "@tailwindcss/vite": "4.1.18",
     "astro": "^6.1.2",
     "pagefind": "^1.4.0",
     "svelte": "^5.55.0",
-    "tailwindcss": "^4.2.2"
+    "tailwindcss": "4.1.18"
   },
   "devDependencies": {
     "@astrojs/check": "^0.9.0",

